sdk/keyvault/Azure.Security.KeyVault.Keys/src/Model.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using Azure.Core;
+using System;
+using System.IO;
+using System.Text.Json;
+
+namespace Azure.Security.KeyVault.Keys
+{
+    public abstract class Model
+    {
+        internal void Deserialize(Stream content)
+        {
+            using (JsonDocument json = JsonDocument.Parse(content, default))
+            {
+                this.ReadProperties(json.RootElement);
+            }
+        }
+
+        internal ReadOnlyMemory<byte> Serialize()
+        {
+            var writer = new ArrayBufferWriter<byte>();
+
+            var json = new Utf8JsonWriter(writer);
```

Comment:
If serializing things is done frequently, you might consider pooling writers (if you are using a version of the package where they're a class) via ConcurrentBag (or some other pooling technique).  It reduces the GC impact of creating the write buffers.

Of course, then you need a strategy for knowing when to not return an object to the pool...  but maybe "always pool" is fine.